src/Runner.Worker/GlobalContext.cs

```diff
@@ -22,5 +23,6 @@ public sealed class GlobalContext
         public StepsContext StepsContext { get; set; }
         public Variables Variables { get; set; }
         public bool WriteDebug { get; set; }
+        public JToken ContainerHookState { get; set; }
```

Comment:
do we care whether this is a JToken, JObject, or just a string?

Reply:
Thanks for mentioning that, `JObject` is the most precise: State is never a `JArray` or a `JProperty`, and we don't need to stringify it here.

`JToken` works, but it's a bit sloppy as it's just the abstract superclass of all the above.

Reply:
Refactored it, using JObject now